packages/patrol/lib/src/custom_finders/patrol_tester.dart

```diff
@@ -538,33 +584,65 @@ class PatrolTester {
           break;
       }
 
+      final settle = _chooseSettleBehavior(andSettle, settleBehavior);
       final resolvedFinder = await dragUntilVisible(
         finder: finder,
         view: scrollablePatrolFinder.first,
         moveStep: moveStep,
         maxIteration: maxScrolls,
         duration: duration,
-        andSettle: andSettle,
+        settleBehavior: settle,
       );
 
       return resolvedFinder;
     });
   }
 
   Future<void> _performPump({
-    required bool? andSettle,
+    required SettleBehavior? settleBehavior,
     required Duration? settleTimeout,
   }) async {
-    final settle = andSettle ?? config.andSettle;
-    if (settle) {
-      final timeout = settleTimeout ?? config.settleTimeout;
-      await tester.pumpAndSettle(
-        const Duration(milliseconds: 100),
-        EnginePhase.sendSemanticsUpdate,
-        timeout,
+    final settle = settleBehavior ?? config.settleBehavior;
+    final timeout = settleTimeout ?? config.settleTimeout;
+    if (settle == SettleBehavior.maybeSettle) {
+      await pumpAndMaybeSettle(
+        timeout: timeout,
+      );
+    } else if (settle == SettleBehavior.settle) {
+      await pumpAndSettle(
+        timeout: timeout,
       );
     } else {
       await tester.pump();
     }
   }
+
+  SettleBehavior? _chooseSettleBehavior(
+    bool? andSettle,
+    SettleBehavior? settleBehavior,
+  ) {
+    SettleBehavior? settle;
+    if (andSettle == null) {
+      settle = settleBehavior;
+    } else {
+      if (andSettle) {
+        settle = SettleBehavior.settle;
+      } else {
+        settle = SettleBehavior.none;
+      }
+    }
+    return settle;
+  }
+}
+
+/// Defines which pump method should be used.
+enum SettleBehavior {
+  /// When pumping should be performed, [PatrolTester.pump] will be called.
```

Comment:
```suggestion
  /// [PatrolTester.pump] is used when pumping.
  ///
  /// This renders a single frame. If some animations are currently working, they will move forward by a single frame.
```

Reply:
A little more explanation can help users understand the differences. The other 2 should be similarly expanded.

If they want to learn more, they can go to the referenced `[PatrolTester.pumpAndXYZ]`.

Reply:
If they can go to referenced method, then why explain it here at all?

Reply:
So they don't have to go to the doc comment of `PatrolTester.pump`, which in turn redirects them to `WidgetTester.pump`, [which is not necessarily beginner friendly](https://api.flutter.dev/flutter/flutter_test/WidgetTester/pump.html).

That's also how Flutter tends to write their docs (every doc comment should be useful on its own). I like this style.